### PR TITLE
Fixes some dex issues with food and natural weapons.

### LINKED ISSUE #4193: Simple mobs do not reset their attack animations on failure
#### Description of issue
Simple mobs like giant spiders do not reset if they pull back to strike but fail to resolve. This causes them to just creep further and further away from their actual location.

#### Difference between expected and actual behavior
They should reset to neutral regardless of the outcome of their attack.

#### Steps to reproduce
Spawn a giant spider and have it attack a dummy human.

#### Length of time in which bug has been known to occur
Likely since the big AI rewrite a few weeks ago.

#### Client version, Server revision & Game ID
Current dev.

#### Issue bingo

- [X] Issue could be reproduced at least once
- [X] Issue could be reproduced by different players
- [X] Issue could be reproduced in multiple rounds
- [X] Issue happened in a recent (less than 7 days ago) round
- [X] [Couldn't find an existing issue about this](https://github.com/NebulaSS13/Nebula/issues)

### LINKED ISSUE #4194: Simple mobs do not show successful attack animations, but apply attack effects like venom.
#### Description of issue
Title.

#### Difference between expected and actual behavior
Attacks should play a sound and show a message prior to applying attack effects.

#### Steps to reproduce
Giant spiders are a good example case currently. Spawn one and let it nibble you.

#### Length of time in which bug has been known to occur
Probably since AI rewrite.

#### Client version, Server revision & Game ID
Current dev.

#### Issue bingo

- [X] Issue could be reproduced at least once
- [X] Issue could be reproduced by different players
- [X] Issue could be reproduced in multiple rounds
- [X] Issue happened in a recent (less than 7 days ago) round
- [X] [Couldn't find an existing issue about this](https://github.com/NebulaSS13/Nebula/issues)

### PR DESCRIPTION
**Pull request by MistakeNot4892**
- Fixes #4193
- Fixes #4194
- Fixes drake food issue in #4191

### CODE REVIEW

File: code/modules/mob/animations.dm [hunk 1]
```diff
@@ -186,3 +186,7 @@
 	else
 		animate(src, pixel_x = pixel_x + pixel_x_diff, pixel_y = pixel_y + pixel_y_diff, time = windup_time-2)
 		animate(pixel_x = default_pixel_x, pixel_y = default_pixel_y, time = 2)
+
+// reset wherever the attack animation got us to.
+/atom/movable/proc/reset_windup_animation()
```
Comment (out-of-phaze) [inline code/modules/mob/animations.dm]: Uh, what's the difference between this and `reset_offsets(anim_time = 2)` (the default)? It just seems like this is less robust? Does it interact with grabs properly?

File: code/modules/reagents/reagent_containers/food.dm [hunk 2]
```diff
@@ -22,6 +22,7 @@
 	center_of_mass = @'{"x":16,"y":16}'
 	w_class = ITEM_SIZE_SMALL
 	abstract_type = /obj/item/chems/food
+	needs_attack_dexterity = DEXTERITY_NONE
```
Comment (out-of-phaze) [inline code/modules/reagents/reagent_containers/food.dm:25]: I don't like this. We confirmed drakes can eat foods fine by using it in-hand, which I thought was the intended way for them to do it? This only affects clicking themselves (and other things) with food, and will undoubtedly have other side effects.

**Review by out-of-phaze — CHANGES_REQUESTED** [review]

File: code/modules/reagents/reagent_containers/food.dm (hunk 2 above)
Comment (MistakeNot4892) [reply code/modules/reagents/reagent_containers/food.dm:25]: This is fixing them not being able to eat food inhand.

Comment (MistakeNot4892) [reply code/modules/reagents/reagent_containers/food.dm:25]: Either that or Greenjoe misreported the issue.

Comment (out-of-phaze) [reply code/modules/reagents/reagent_containers/food.dm:25]: it was misreported, we confirmed it on discord. using it inhand worked, clicking yourself is what didn't and that's intended afaik

Comment (MistakeNot4892) [reply code/modules/reagents/reagent_containers/food.dm:25]: Per Discord I think I was misunderstood re: intentionality. I don't think it's good UX to arbitrarily have one method work and the other not for something as fundamental as eating. With respect to side effects, I can't think of anything offhand other than drakes feeding food to people, which is a bit whatever.

File: code/modules/mob/animations.dm (hunk 1 above)
Comment (MistakeNot4892) [reply code/modules/mob/animations.dm]: It was just extracting the existing line. I'll swap it out for reset_offsets() I guess.

File: code/modules/reagents/reagent_containers/food.dm (hunk 2 above)
Comment (out-of-phaze) [reply code/modules/reagents/reagent_containers/food.dm:25]: in that case yeah, this is fine

**Review by out-of-phaze — DISMISSED** [review]

File: code/_onclick/other_mobs.dm [hunk 3]
```diff
@@ -79,6 +79,7 @@
 	if(a_intent == I_HELP || !attacking_with)
 		return A.attack_animal(src)
 
+	a_intent = I_HURT
```
Comment (out-of-phaze) [inline code/_onclick/other_mobs.dm:81]: already exists on line 77

Comment (MistakeNot4892) [reply code/_onclick/other_mobs.dm:81]: Line 77 is actually in error. Resolved now anyway.

**Review by out-of-phaze — APPROVED** [review]
still 24h under conflict resolution rule, yay

**Pull request merged**